In [2]:
import os
import subprocess
import pandas as pd
import glob

In [2]:
pwd


'/Users/gracer/Desktop/fsl_course_data/script'

In [3]:
basepath = '/Users/gracer/Desktop/fsl_course_data'

In [6]:
os.path.exists(os.path.join(basepath, 'fmri1','fluency_task'))

True

In [9]:
for x in glob.glob(os.path.join(basepath, 'fmri1','fluency_task','*.nii.gz')):
    print(x)

/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/fmri.nii.gz
/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/structural_brain.nii.gz
/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/fmri_brain.nii.gz
/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/fmri_brain_mask.nii.gz
/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/structural.nii.gz


raw
-- sub-001> fmri.nii.gz, struct.nii.gz
-- sub-002> fmri.nii.gz, struct.nii.gz

In [19]:
SUBNUM = []
for z in glob.glob(os.path.join(basepath,'fmri1','raw_data','sub-*')):
    y = z.split('/')[-1].split('-')[-1]
    SUBNUM.append(y)

002
001


In [20]:
SUBNUM

['002', '001']

In [21]:
SUBNUM = ['002','001']

In [4]:
cmd = 'fslnvol %s'%subnum


NameError: name 'subnum' is not defined

In [29]:
for sub in SUBNUM:
    SUB = os.path.join(basepath, 'fmri1','raw_data','sub-%s'%sub,'fmri.nii.gz')
    cmd = 'fslnvols %s'%SUB
    cmd_split = cmd.split(' ')
    subprocess.run(cmd_split)

106
106


# Warm up 10/18/22
Based on what you know and information below. Get the TR value for the fmri_brain.nii.gz scan using subprocess. You will want to use the check_output function in subprocess. This will allow you to assign the output to a variable in python. 
```
X = subprocess.check_output(['ls','-a','<path to direcrory>'] 
```
The above would assign all the output from the ls command using the -a flag to the variable X
Check_output returns bytes (not STRINGS). So we also need to decode the byte into a string
```
Y = X.decode('utf-8')
```
Now Y is a string. 

You final TR should be a string with only the value of the TR, no extra characters

 You will use the following command:
 ```
 fslval <functional scan> pixdim4
 ```

In [1]:
import os
import subprocess
import pandas as pd
import glob

In [5]:
func_path = os.path.join(basepath, 'fmri1', 'fluency_task','fmri_brain.nii.gz')

In [6]:
os.path.exists(func_path)

True

In [17]:
# fslval func_path pixdim4 <- original command
# option 1
cmd = 'fslval %s pixdim4'%func_path
print(cmd)
cmd_split = cmd.split(' ')
print(cmd_split)
trs_alt = subprocess.check_output(cmd_split)
print(trs_alt)
trs_alt = trs_alt.decode('utf-8')
print(trs_alt)
final_trs = trs_alt.split(' ')[0]
print(final_trs)
# option 2
trs = subprocess.check_output(['fslval','%s'%(func_path),'pixdim4'])
# change the byte to string
trs = trs.decode('utf-8')
# get the number only
trs = trs.split(' ')[0]
print(trs)

fslval /Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/fmri_brain.nii.gz pixdim4
['fslval', '/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/fmri_brain.nii.gz', 'pixdim4']
b'4.200000 \n'
4.200000 

4.200000
4.200000


## Notes to self
* naming convention = sub-001
* subnum == the subject number
* OUTPUT = path to the output file = os.path.join(basepath,'level1','sub-%s'%subnum)
* TR = 4.2
* nvols = 106
* INPUT = path to the functional data

# Dictionary

In [18]:
inpt = os.path.join(basepath, 'fmri1','fluency_task','fmri_brain.nii.gz')

In [23]:
os.path.exists(inpt)

True

In [24]:
repl_dict = {}

In [25]:
repl_dict.update({'INPUT':inpt})

In [26]:
repl_dict

{'INPUT': '/Users/gracer/Desktop/fsl_course_data/fmri1/fluency_task/fmri_brain.nii.gz'}

In [27]:
outpath = os.path.join(basepath,'derivatives','sub-001.feat')

# For next time 10/25/22
## To do
Update repl_dict with all of your variables that in skeleton.fsf